File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/visual/PlayerEdit.java
package net.ccbluex.liquidbounce.features.module.modules.visual;

import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.ModuleCategory;
import net.ccbluex.liquidbounce.features.module.ModuleInfo;
import net.ccbluex.liquidbounce.value.BoolValue;
import net.ccbluex.liquidbounce.value.FloatValue;
import net.ccbluex.liquidbounce.value.IntegerValue;
import net.ccbluex.liquidbounce.value.ListValue;

@ModuleInfo(name = "PlayerEdit", description = "Edit the player.", category = ModuleCategory.VISUAL)
public class PlayerEdit extends Module {

    public static BoolValue editPlayerSizeValue = new BoolValue("EditPlayerSize", false);
    public static FloatValue playerSizeValue = new FloatValue("PlayerSize", 0.5f,0.01f,5f);
    public static BoolValue rotatePlayer = new BoolValue("RotatePlayer", false);
    public static BoolValue customModel = new BoolValue("CustomModel", false);
    public static ListValue mode = new ListValue("Mode", new String[]{"Freddy", "Rabbit", "Amogus"}, "Freddy");
    public static IntegerValue bodyColorR = new IntegerValue("BodyR", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue bodyColorG = new IntegerValue("BodyG", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue bodyColorB = new IntegerValue("BodyB", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue eyeColorR = new IntegerValue("EyeR", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue eyeColorG = new IntegerValue("EyeG", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue eyeColorB = new IntegerValue("EyeB", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue legsColorR = new IntegerValue("LegsR", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue legsColorG = new IntegerValue("LegsG", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static IntegerValue legsColorB = new IntegerValue("LegsB", 255,0,255, () -> mode.get().equalsIgnoreCase("Amogus"));
    public static BoolValue baby = new BoolValue("Baby", false);
    public static BoolValue onlyMe = new BoolValue("OnlyMe", false);
    public static BoolValue onlyOther = new BoolValue("OnlyOther", false);
}

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/visual/PlayerEdit.kt
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge by LiquidBounce.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.features.module.modules.visual

import me.zywl.fdpclient.event.EventTarget
import me.zywl.fdpclient.event.MotionEvent
import net.ccbluex.liquidbounce.features.module.Module
import net.ccbluex.liquidbounce.features.module.ModuleCategory
import net.ccbluex.liquidbounce.features.module.ModuleInfo
import me.zywl.fdpclient.value.impl.BoolValue
import me.zywl.fdpclient.value.impl.FloatValue

/**
 * The type Player edit.
 */
@ModuleInfo(name = "PlayerEdit", description = "", category = ModuleCategory.VISUAL)
class PlayerEdit : Module() {

    companion object {
        @JvmField
        val editPlayerSizeValue = BoolValue("PlayerSize", false)

        /**
         * The constant playerSizeValue.
         */
        @JvmField
        val playerSizeValue = FloatValue("PlayerSize", 1.5f, 0.5f, 2.5f)

        /**
         * The constant rotatePlayer.
         */
        @JvmField
        val rotatePlayer = BoolValue("PlayerRotate", true)

        @JvmField
        val xRot = FloatValue("X-Rotation", 90.0f, -180.0f, 180.0f)

        @JvmField
        val yPos = FloatValue("Y-Position", 0.0f, -5.0f, 5.0f)
    }

    @EventTarget
    fun onMotion(event: MotionEvent?) {
        if (editPlayerSizeValue.get()) mc.thePlayer.eyeHeight =
            playerSizeValue.get() + 0.62f else mc.thePlayer.eyeHeight = mc.thePlayer.defaultEyeHeight
    }

    override fun onDisable() {
        mc.thePlayer.eyeHeight = mc.thePlayer.defaultEyeHeight
    }
}